lib/src/source/hosted.dart

```diff
@@ -860,16 +863,27 @@ class HostedSource extends CachedSource {
       var archivePath =
           p.join(tempDirForArchive, '$packageName-$version.tar.gz');
       var response = await withAuthenticatedClient(
-          cache,
-          Uri.parse(description.url),
-          (client) => client.send(http.Request('GET', url)));
+          cache, Uri.parse(description.url), (client) {
+        var req = http.Request('GET', url);
+        // TODO: How can we properly check if a request is GCS bound, so only
+        // those requests get this header? (for checksum validation)
+        req.headers['Cache-Control'] = 'no-transform';
```

Comment:
How can we properly check if a request is GCS bound, so only those requests get this header?

Reply:
Hmm - could we just always send this header?
I think for the coming content-hashing: https://github.com/dart-lang/pub/pull/3482 any transformations to the archive (on any server) would be problematic.

Furthermore we can validate the crc32 checksum whenever it is provided, it doesn't strictly have to be from gcs...

We could even update the https://github.com/dart-lang/pub/blob/master/doc/repository-spec-v2.md to mention the header, and explain that it is checked.

Reply:
I'd think it'd be safe to always send it.

But also unnecessary, I don't think we have any intermediate proxy, and if we did we shouldn't allow to do transformations. No need to compress a `tar.gz` again :see_no_evil:

Reply:
So, I wrote this under the assumption that [adding this header would impair performance](https://stackoverflow.com/a/20197125) for requests that are bound for custom hosts. But it sounds like this header is desired regardless for the upcoming content hashing feature.

I can keep the header and add a comment/note?

Also, let me know if you'd like me to update the repository spec to include this header. 🫡